.github/workflows/add-labels.yml

```
@@ -0,0 +1,54 @@
+name: add labels for some cases
+
+on: pull_request
+
+jobs:
+  reviewapp:
+    runs-on: ubuntu-latest  # windows-latest | macos-latest
+    name: grant 'reviewapps' label if there are any changes in PR's source code.
+    if: ${{ ! startsWith(github.head_ref, 'renovate/') }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get changed files
+        id: check-paths-ignore
+        uses: tj-actions/changed-files@v42
+        with:
+          files_ignore: |
+            .github/**/*.yml
+            **.md
+
+      # to trigger other Actions caused by adding reviewapp Label
+      - name: Generate token
+        if: steps.check-paths-ignore.outputs.any_changed == 'true'
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.PRIVATE_KEY }}
+
+      - name: Labeling 'reviewapps' to PR
+        if: steps.check-paths-ignore.outputs.any_changed == 'true'
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ steps.generate_token.outputs.token }}
+          result-encoding: string
+          script: |
+            const targetLabel = 'reviewapps';
+            issue = await github.rest.issues.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+            });
+            flag = false;
+            issue.data.labels.filter(label => {
+              if (label.name == targetLabel) { flag = true; };
+            });
+            if (!flag) {
```

```suggestion
            const hasLabel = issue.data.labels.some(label => label.name === targetLabel);
            if (!hasLabel) {
```

細かいんですが、書き方がJavaScriptっぽくなかったので。
（なお、配列の要素を巡回したいだけで値は使わないなら、filterよりもforEachの方が良いです。今回は、さらにsomeの方が良さそうです）